0000-product.md

```diff
@@ -0,0 +1,470 @@
+belong to an organization in Pike state, otherwise the transaction is invalid.
+* The agent must have the permission can_create_product for the organization,
+otherwise the transaction is invalid.
+* If the product_namespace is GS1, the organization must contain a GS1 Company Prefix in its metadata (gs1_company_prefixes), and the prefix must match the company prefix in the product_id, which is a gtin if GS1, otherwise the transaction is invalid.  
+* The properties must be valid for the product_namespace. For example, if the product is GS1 product, its properties must only contain properties that are includedin the GS1 Schema. If it includes a property not in the GS1 Schema the transaction is invalid.  _The base GS1 schema will be defined in a future RFC._
+
+The product will be set in state.
+
+The inputs for ProductCreateAction must include:
+
+* Address of the Agent submitting the transaction Address of the Organization
+* the Product is being created for Address of the Product Namespace Schema the
+* product’s properties must match Address of the Product to be created
+
+The outputs for ProductCreateAction must include:
+
+* Address of the Product created
+
+### ProductUpdateAction
+
+ProductUpdateAction updates an existing product in state. The transaction should
+be submitted by an agent, identified by its signing key, acting on behalf of an
+organization that corresponds to the owner in the product being updated.
+(Organizations and agents are defined by the Pike smart contract.)
+
+``` 
+message ProductUpdateAction { 
+    enum Product_Namespace { 
+            UNSET_NAMESPACE = 0;
+            GS1 = 1; 
+    }
+    // product_namespace and product_id are used in deriving the state address
+    Product_Namespace product_namespace = 1; 
+    string product_id = 2;
+    // this will replace all properties currently defined
+    repeated PropertyValues properties = 4; 
+} 
+```
+
+Validation requirements:
+
+* If a product with product_id does not exist the transaction is invalid.  The
+* signer of the transaction must be an agent in the Pike state and must
+belong to an organization in Pike state, otherwise the transaction is invalid.
+* The owner in the product must match the organization that the agent belongs
+* to, otherwise the transaction is invalid.  
+* The agent must have the permission can_update_prouduct for the organization,
+otherwise the transaction is invalid.
+* The new properties must be valid for the product_namespace. 
+* For example, if the product is GS1 product, its properties must only contain properties that are included in the GS1 Schema. If it includes a property not in the GS1 Scheme the transaction is invalid.
+
+The properties in the product will be swapped for the new properties and the
+updated product will be set in state.
+
+The inputs for ProductUpdateAction must include:
+
+* Address of the Agent submitting the transaction Address of the Organization
```

Comment:
This bullet section is messed up. Several points are now running together and do not make sense.